### PR TITLE
feat: add locator timings

### PR DESCRIPTION
**Pull request by wswebcreation**
The goal of this PR is to give people insights on the time it will take to locate elements with a certain locator strategy. It will add:
- a button in the inspector window to get times of finding elements with a certain locator strategy
- a spinner for waiting
- find element times in the _Find by_ table

**Android example**
appium-times-android

**iOS example:**
appium-times-ios

### CODE REVIEW

File: app/main/appium-method-handler.js [hunk 1]
```diff
@@ -64,7 +64,11 @@ export default class AppiumMethodHandler {
   }
 
   async fetchElement (strategy, selector) {
+    const start = process.hrtime.bigint();
```
Comment (mykola-mokhnach) [inline app/main/appium-method-handler.js:67]: Please use the Timer class from appium-support lib

Comment (mykola-mokhnach) [reply app/main/appium-method-handler.js:67]: https://github.com/appium/appium-support/blob/b90d85fde1c1ce2aee0a2511e5bbff5cc11c1549/lib/net.js#L155

File: app/renderer/components/Inspector/SelectedElement.js [hunk 2]
```diff
@@ -140,30 +156,39 @@ class SelectedElement extends Component {
...
             >
               {t('Clear')}
             </Button>
+            <Tooltip title={t('getTimes')}>
+              <Button
+                disabled={isDisabled}
+                id='btnGetTimings'
+                onClick={() => getFindElementsTimes(findDataSource)}
+              >
+                {t('Get Times')}
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/SelectedElement.js:186]: Is this string present in resources?

**Comment by mykola-mokhnach** [comment]
I think from UX perspective it is not very obvious that Get Times button should fill up these cells. Would it be possible to put such button or a link directly into the cell where the resulting number should appear?

File: app/main/appium-method-handler.js (hunk 1 above)
Comment (wswebcreation) [reply app/main/appium-method-handler.js:67]: Thanks, will do

File: app/renderer/components/Inspector/SelectedElement.js (hunk 2 above)
Comment (wswebcreation) [reply app/renderer/components/Inspector/SelectedElement.js:186]: No, will add it

**Comment by wswebcreation** [comment]
> I think from UX perspective it is not very obvious that Get Times button should fill up these cells. Would it be possible to put such button or a link directly into the cell where the resulting number should appear?

@mykola-mokhnach 

I've added a button to make it more clear like this, the tool tip is still present.


image

image

**Comment by mykola-mokhnach** [comment]
I think `Timing` would be a more precise term here. `Get Time` has a bit different meaning to me (although I'm not a native speaker, so it might be a wrong assumption)

**Comment by wswebcreation** [comment]
> I think `Timing` would be a more precise term here. `Get Time` has a bit different meaning to me (although I'm not a native speaker, so it might be a wrong assumption)

Change it to this, including the tool tip

image

**Comment by wswebcreation** [comment]
Tnx for merging @KazuCocoa 

Can we get a new release in the coming days with those 2 new features?

**Comment by KazuCocoa** [comment]
Perhaps yes.
Once bump current version like https://github.com/appium/appium-desktop/pull/1414/commits/f0ffcddc7598f268fdf342df04b6e2ab45f5f7eb and run a CI job, the new one will override https://github.com/appium/appium-desktop/releases/tag/v1.18.0-beta.0 (Or we should increment the version)
Let me try

**Comment by KazuCocoa** [comment]
@wswebcreation 
Current master is available from https://github.com/appium/appium-desktop/releases/tag/v1.18.0-beta.1 as pre release

**Comment by wswebcreation** [comment]
@KazuCocoa 

Thanks!! It's working, just one question, can you also add this PR https://github.com/appium/appium-desktop/pull/1448 in the description? The `Get Timings` is also merged and in this release

**Comment by KazuCocoa** [comment]
👍 added

**Pull request merged**